crates/sc-consensus-subspace/src/lib.rs

```diff
@@ -702,6 +705,39 @@ where
                 .ok_or(Error::<Block>::MissingSalt(hash))?
                 .salt;
 
+            let parent_block_id = BlockId::Hash(*block.header.parent_hash());
```

Comment:
As I mentioned in https://github.com/subspace/subspace/pull/302 and during calls this week, this is the wrong place to check such things if you rely on parent block being available. This must be stateless check (also see comment on line 689 above). This needs to be checked during block import instead.